### PR TITLE
Fix readme examples

### PR DESCRIPTION
**Pull request by laat**
formatting parse result to encode information about keys
```
//=> {foo: ['1', '2', '3']}
//=> {'foo[]': [1, 2, 3]}
```

Fixed some other small issues

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -110,7 +110,7 @@ Default: `false`
 
 ```js
 queryString.parse('foo[]=1&foo[]=2&foo[]=3', {parseNumbers: true});
-//=> foo: [1, 2, 3]
+//=> {'foo[]': [1, 2, 3]}
```
Comment (laat) [inline readme.md]: behaves like this today, is this expected?

Comment (sindresorhus) [inline readme.md]: I think we need a simpler example here. Maybe just `queryString.parse('foo=1', {parseNumbers: true});`

**Comment by sindresorhus** [comment]
Thanks for catching these mistakes. Can you also update the examples in https://github.com/sindresorhus/query-string/blob/master/index.d.ts ?

**Comment by laat** [comment]
will do

**Pull request merged**